crates/libgraphql-core/src/test/golden_tests/test_runner.rs

```diff
@@ -0,0 +1,632 @@
+    let test_name = format!("{}/schema", test_case.name);
+
+    // Build schema from all schema files
+    let mut builder = SchemaBuilder::new();
+
+    for schema_path in &test_case.schema_paths {
+        builder = match builder.load_file(schema_path) {
+            Ok(b) => b,
+            Err(e) => {
+                return GoldenTestResult {
+                    test_name,
+                    passed: false,
+                    error_message: Some(format!("Expected: Valid schema\nGot: {e:?}")),
+                    file_path: schema_path.clone(),
+                    file_snippet: None,
+                };
+            }
+        };
+    }
+
+    match builder.build() {
+        Ok(_) => GoldenTestResult {
+            test_name,
+            passed: true,
+            error_message: None,
+            file_path: test_case.schema_paths[0].clone(),
+            file_snippet: None,
+        },
+        Err(e) => {
+            let error_str = format!("{e:?}");
+            let file_path = test_case.schema_paths[0].clone();
+            let snippet = extract_snippet_with_error_marker(&file_path, 3).ok();
+
+            GoldenTestResult {
+                test_name,
+                passed: false,
+                error_message: Some(format!("Expected: Valid schema\nGot: {error_str}")),
+                file_path,
+                file_snippet: snippet,
+            }
+        }
+    }
+}
+
+/// Test a single invalid schema (single or multi-file)
+fn test_invalid_schema(test_case: &GoldenTestCase) -> GoldenTestResult {
+    let test_name = format!("{}/schema", test_case.name);
+
+    // Build schema from all schema files
+    let mut builder = SchemaBuilder::new();
+
+    for schema_path in &test_case.schema_paths {
+        builder = match builder.load_file(schema_path) {
+            Ok(b) => b,
+            Err(e) => {
+                // Early error during loading - check if it matches expected patterns
+                let error_str = format!("{e:?}");
+                let errors = [error_str.clone()];
+
+                if !test_case.schema_expected_errors.is_empty() {
+                    let all_match = test_case
+                        .schema_expected_errors
+                        .iter()
+                        .all(|pattern| errors.iter().any(|e| e.contains(pattern)));
+
+                    if all_match {
+                        return GoldenTestResult {
+                            test_name,
+                            passed: true,
+                            error_message: None,
+                            file_path: schema_path.clone(),
+                            file_snippet: None,
+                        };
+                    }
+                }
+
+                // Error occurred but didn't match expected
+                return GoldenTestResult {
+                    test_name,
+                    passed: true, // Schema failed as expected
```

Comment:
If an error occurred but didn't match the expected set of errors, shouldn't we set `passed: false` here?